frontend/src/metabase/query_builder/components/notebook/lib/steps.js

```diff
@@ -110,7 +110,14 @@ const STEPS: StepDefinition[] = [
     revert: (query, index) => query.removeJoin(index),
     clean: (query, index) => {
       const join = query.joins()[index];
-      return !join || join.isValid() ? query : query.removeJoin(index);
+      if (!join || join.isValid()) {
+        return query;
+      }
+      const cleanJoin = join.clean();
+      if (!cleanJoin.isValid()) {
```

Comment:
this logic might be totally correct, but I'm finding it confusing to understand. Let me put how I'm reading it into english:

1. if a given join either doesn't exist or is valid, do nothing to the query.
2. otherwise (if a join exists and is invalid), create a cleaned instance of the join
3. if the cleaned join still isn't valid, replace the original, invalid join with the cleaned, invalid join
4. if the cleaned join is now valid, remove the original, invalid join

is that correct? I find that the steps to this function, in particular steps 3 and 4, don't really follow how you'd expect

Reply:
Ouch, you're absolutely right, if the clean join is valid, we should replace the original one with it. Otherwise, we should just remove the whole join. Thank you so much for being so attentive